WordPress/gutenberg
# Post Editor: Update publish flow

## What?
Part of: https://github.com/WordPress/gutenberg/issues/59878

This PR aims to update the publishing flow in post editor, so we can later unify it with site editor.

### Tasks

1. Update the post status component design and use it in post editor

Panel               |  Post status label
-----------------------------------------|-------------------------
![Panels](https://github.com/WordPress/gutenberg/assets/846565/4879dfe9-f54e-49d4-9920-4d504cc64b24) |  <img width="896" alt="status" src="https://github.com/WordPress/gutenberg/assets/846565/89221af8-fe56-47d2-8590-44c243dbf63e">

From the above designs I'm not sure we can add another control(publish date) in between the radio control options. --cc @jameskoster @mirka 


2. Update `publish` button

- Match `save` label and save panel when multiple entities have changed
- handle pending status/button
- handle switch to draft status/button properly
- remove `visibility` panel


 ## Testing Instructions

Testing should be thorough including any possible flow. Examples:
1. Make changes in a post by updating the status directly from the panel, like from `published to draft`, etc.. 
2. Do the same as above with other content changes and other entities' changes
3. Make changes to a `draft or pending` post and use `save as draft` or `save as pending` buttons in combination with other changes and changes in other entities.
4. Do the same as above and use the main `save/publish` button
5. Make changes only to other entities(pattern) and save either to that page or by returning to the post editor of the main entity(page).
6. I guess other flows that I'm missing too..
